sql/catalyst/src/main/scala/org/apache/spark/sql/AnalysisException.scala

```diff
@@ -54,10 +72,34 @@ class AnalysisException protected[sql] (
       messageParameters: Array[String],
       origin: Origin) =
     this(
-      SparkThrowableHelper.getMessage(errorClass, messageParameters),
+      SparkThrowableHelper.getMessage(errorClass, None, messageParameters),
+      line = origin.line,
+      startPosition = origin.startPosition,
+      errorClass = Some(errorClass),
+      errorSubClass = None,
+      messageParameters = messageParameters)
+
+  def this(
+            errorClass: String,
+            errorSubClass: String,
+            messageParameters: Array[String]) =
```

Comment:
Just dropping a comment for a nit. maybe would have to be four spaces per https://github.com/databricks/scala-style-guide#spacing-and-indentation:

```scala
  def this(
      errorClass: String,
      errorSubClass: String,
      messageParameters: Array[String]) =
```